source/monitoring/cluster-monitoring.txt

```diff
@@ -0,0 +1,135 @@
+.. _php-cluster-monitoring:
+
+==================
+Cluster Monitoring
+==================
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: code example, server, topology
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecols
+
+Overview
+--------
+
+This guide shows you how to use the {+php-library+} to monitor topology
+events in a MongoDB instance, replica set, or sharded cluster. The
+driver creates topology events, also known as Server Discovery and
+Monitoring (**SDAM**) events, when there are any changes in the state of the
+MongoDB instance or cluster that you are connected to.
+
+You might use information about topology events in your
+application to understand cluster changes, assess cluster health, or
+perform capacity planning.
+
+Subscribe to Events
+-------------------
+
+You can access details about SDAM events by subscribing to them
+in your application. To subscribe to any event, create a class that
+implements the ``MongoDB\Driver\Monitoring\SDAMSubscriber`` interface,
+then use the ``MongoDB\Client::addSubscriber()`` method to register the
+event subscriber with your ``MongoDB\Client`` instance.
+
+The following code creates the ``MySubscriber`` class that implements
+``SDAMSubscriber`` to output a message when a ``ServerOpeningEvent`` is
+generated by the server:
+
+.. literalinclude:: /includes/monitoring/sdam.php
+   :start-after: start-mysubscriber
+   :end-before: end-mysubscriber
+   :language: php
+   :copyable:
+   :dedent:
+
+.. note::
+
+   As shown in the preceding code, you must implement all of the methods
+   of ``SDAMSubscriber``, even for events you are not subscribing to.
+   You can implement these methods with empty bodies so that the library
+   does not generate any messages for these events.
+
+Then, the following code uses the ``addSubscriber()`` method to register
+``MySubscriber`` with the client:
+
+.. literalinclude:: /includes/monitoring/sdam.php
+   :start-after: start-add-sub
+   :end-before: end-add-sub
+   :language: php
+   :copyable:
+   :dedent:
+
+When you run the application, your subscriber records the SDAM event and
+outputs messages such as the following:
+
+.. code-block:: none
+   :copyable: false
+
+   Server opening on ac-rmuag0v-shard-00-00.gh0qg50.mongodb.net:27017
+   Server opening on ac-rmuag0v-shard-00-01.gh0qg50.mongodb.net:27017
+   Server opening on ac-rmuag0v-shard-00-02.gh0qg50.mongodb.net:27017
+
+Event Descriptions
+------------------
+
+You can subscribe to the following SDAM events by implementing the
+corresponding method from the ``SDAMSubscriber`` interface:
+
+.. list-table::
+   :widths: 35 65 
+   :header-rows: 1
+
+   * - Event Name
+     - Description
+
+   * - ``ServerChangedEvent``
+     - Created when an instance state changes, such as from secondary to
```

Comment:
Optional: Could remove "Created when" from the start of every event (probably along with a new column header name)

Reply:
I can't think of a good way to adjust the column header, as renaming it "Situation that Triggers Event" seems worse than keeping the "Created when"s in

Reply:
In the docs for [ServerChangedEvent](https://www.php.net/manual/en/class.mongodb-driver-monitoring-serverchangedevent.php) we say the event class "encapsulates information about a changed server description".

Here, the table is presenting the event as the thing that happened instead of just an object containing some information about a thing that happened.

The "Condition" column in [Events API](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring-logging-and-monitoring.md#events-api-1) from the driver spec might be helpful. Note that the spec includes "Description" in the name of these events, where PHP just calls it ServerChangedEvent. I think that was due to us copying the shorter name from our libmongoc dependency.

I might suggest:

> When the server description changes, such as the server's type changing from secondary to primary.